store/mongo/datastore_mongo.go

```diff
@@ -245,6 +245,14 @@ func (db *DataStoreMongo) PutDeviceAuth(ctx context.Context, dev *model.DeviceAu
 	return nil
 }
 
+func (db *DataStoreMongo) InsertDeviceAuth(ctx context.Context, dev *model.DeviceAuth) error {
+
+	dev.ID = model.AuthID(bson.NewObjectId().Hex())
+	dev.DeviceId = model.DeviceID(bson.NewObjectId().Hex())
+
+	return db.PutDeviceAuth(ctx, dev)
```

Comment:
to consider: replace PutDeviceAuth with a bare-bones `mgo.Insert`?
PutDeviceAuth is quirky, we will be moving away from it in favor of strict updates and inserts (actually, it's buggy - allows updating a nonexistent authset).